### DISCUSSION TITLE
Generate Stub files in Quarkus/Java from proto files

### DISCUSSION THREAD

**Discussion by manuelwallrapp** [tektoncd/results#370] [Q&A] [answered]
Hi

I need a little help with the proto files to generate the stubs in Java. We are using Quarkus. If I use a hello world example proto files my Maven build generates the grpc stubs without issues. If I process one or more of the tekton-results proto files, the stub generation fails. Can you describe how I can generate the Grpc Stubs for Java? Which files to use?

Kind regards

Manu

**Comment by manuelwallrapp** [comment, marked as answer]
Ok I found out, needed to also copy the files:
google/api/field_behavior.proto
google/api/resource.proto

from here: https://github.com/googleapis/googleapis/tree/master/google/api

to the local proto directory in the corresponding subdirectories: proto/google/api

After then I could generate the GPRC Stubs with Quarkus/Java.

**Reply by adambkaplan** [reply]
It looks like the grpc google APIs are also available through maven central - adding this to your `pom.xml` may also fix the problem:

```
<!-- https://mvnrepository.com/artifact/com.google.api-client/google-api-client-protobuf -->
<dependency>
    <groupId>com.google.api-client</groupId>
    <artifactId>google-api-client-protobuf</artifactId>
    <version>2.2.0</version>
</dependency>
```

**Reply by manuelwallrapp** [reply]
Thank you @adambkaplan sadly it doesn't work. In the end I copied following files into the proto/google/api folder and the quarkus build works:
annotations.proto
client.proto
field_behavior.proto
http.proto
launch_stage.proto
resource.proto

Of course I would also prefer to have only a maven dependency.

**Reply by adambkaplan** [reply]
I would raise this with the gRPC community then - this feels like a bug.

**Comment by adambkaplan** [comment]
Closing this discussion as "answered" :heavy_check_mark:
